iexec-sms-library/src/test/java/com/iexec/sms/api/SmsClientProviderTests.java

```diff
@@ -0,0 +1,304 @@
+                        " [chainDealId: " + CHAIN_DEAL_ID_1 + ", chainTaskId: " + CHAIN_TASK_ID_1 +"]",
+                e.getMessage());
+    }
+    // endregion
+
+    // region getSmsUrlForUninitializedTask
+    @Test
+    void shouldGetSmsUrlForUninitializedTaskWhenTaskFound() {
+        Mockito.when(iexecHubService.getChainDeal(CHAIN_DEAL_ID_1)).thenReturn(Optional.of(CHAIN_DEAL_1));
+
+        Optional<String> smsUrl = smsClientProvider.getSmsUrlForUninitializedTask(CHAIN_DEAL_ID_1, CHAIN_TASK_ID_1);
+        assertTrue(smsUrl.isPresent());
+        assertEquals(SMS_URL_1, smsUrl.get());
+
+        verify(iexecHubService, times(1)).getChainDeal(CHAIN_DEAL_ID_1);
+        verify(iexecHubService, times(0)).getTaskDescription(any());
+    }
+
+    @Test
+    void shouldGetSmsUrlOnlyOnceForSameUninitializedTask() {
+        Mockito.when(iexecHubService.getChainDeal(CHAIN_DEAL_ID_1)).thenReturn(Optional.of(CHAIN_DEAL_1));
+
+        Optional<String> smsUrlFirstCall = smsClientProvider.getSmsUrlForUninitializedTask(CHAIN_DEAL_ID_1, CHAIN_TASK_ID_1);
+        assertTrue(smsUrlFirstCall.isPresent());
+        assertEquals(SMS_URL_1, smsUrlFirstCall.get());
+
+        verify(iexecHubService, times(1)).getChainDeal(CHAIN_DEAL_ID_1);
+        verify(iexecHubService, times(0)).getTaskDescription(any());
+
+        Optional<String> smsUrlSecondCall = smsClientProvider.getSmsUrlForUninitializedTask(CHAIN_DEAL_ID_1, CHAIN_TASK_ID_1);
+        assertTrue(smsUrlSecondCall.isPresent());
+        assertEquals(SMS_URL_1, smsUrlSecondCall.get());
+
+        verify(iexecHubService, times(1)).getChainDeal(CHAIN_DEAL_ID_1);
+        verify(iexecHubService, times(0)).getTaskDescription(any());
+    }
+
+    @Test
+    void shouldGetSmsUrlTwiceForDifferentUninitializedTasks() {
+        Mockito.when(iexecHubService.getChainDeal(CHAIN_DEAL_ID_1)).thenReturn(Optional.of(CHAIN_DEAL_1));
+        Mockito.when(iexecHubService.getChainDeal(CHAIN_DEAL_ID_2)).thenReturn(Optional.of(CHAIN_DEAL_2));
+
+        Optional<String> smsUrlFirstCall = smsClientProvider.getSmsUrlForUninitializedTask(CHAIN_DEAL_ID_1, CHAIN_TASK_ID_1);
+        assertTrue(smsUrlFirstCall.isPresent());
+        assertEquals(SMS_URL_1, smsUrlFirstCall.get());
+
+        verify(iexecHubService, times(1)).getChainDeal(CHAIN_DEAL_ID_1);
+        verify(iexecHubService, times(0)).getTaskDescription(any());
+
+        Optional<String> smsUrlSecondCall = smsClientProvider.getSmsUrlForUninitializedTask(CHAIN_DEAL_ID_2, CHAIN_TASK_ID_2);
+        assertTrue(smsUrlSecondCall.isPresent());
+        assertEquals(SMS_URL_2, smsUrlSecondCall.get());
+
+        verify(iexecHubService, times(1)).getChainDeal(CHAIN_DEAL_ID_2);
+        verify(iexecHubService, times(0)).getTaskDescription(any());
+    }
+
+    @Test
+    void shouldNotGetSmsUrlForTaskWhenDealNotFound() {
+        Mockito.when(iexecHubService.getChainDeal(CHAIN_DEAL_ID_1)).thenReturn(Optional.of(CHAIN_DEAL_1));
```

Comment:
CHAIN_DEAL_1 has an URL
Condition on line 298 fails

Reply:
Good catch, thanks! Fixed in https://github.com/iExecBlockchainComputing/iexec-sms/pull/116/commits/6f2da4d351e291f2a9b3714784b1d0e44728b060.